File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/player/ResolvedPlayer.java
package top.likoslupus.cellulosesz.api.player;

import org.jspecify.annotations.Nullable;
import top.likoslupus.cellulosesz.api.platform.CellPlayer;

import java.util.Optional;
import java.util.UUID;

public record ResolvedPlayer(
        ResolvedPlayerState state,
        @Nullable UUID uuid,
        String name,
        @Nullable CellPlayer onlinePlayer,
        boolean vanished
) {

    public Optional<UUID> optionalUuid() {
        return Optional.ofNullable(uuid);
    }

    public Optional<CellPlayer> online() {
        return Optional.ofNullable(onlinePlayer);
    }

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/player/ResolvedPlayer.kt
package top.likoslupus.cellulosesz.api.player

import top.likoslupus.cellulosesz.api.platform.CellPlayer
import java.util.*

@JvmRecord
public data class ResolvedPlayer(
    public val state: ResolvedPlayerState,
    public val uuid: UUID?,
    public val name: String,
    public val onlinePlayer: CellPlayer?,
    public val vanished: Boolean
)
